design_drafts/flexible_joint_states_msg.md

```diff
@@ -34,28 +56,122 @@ float64[] velocity
 float64[] effort
 ```
 
-## First proposal
+The following proposal reflects a map-like (similar to what `std::unordered_map` represents in C++ or a hash table in other languages) datatype where not only three statically chosen value-identifier are attached to a joint name, but can be set dynamically.
+
+### Proposal for a Dynamic Joint State Message
 
 ```
 Header header
 
-string[] name
+string[] interface_name
+InterfaceValue[] interface_value
+  string[] value_identifier
+  float64[] value
+```
+
+where `name` stands for a list of joint or hardware interface names, `interface_value` is a vector of a new message, being essentially a key-value pair, e.g. `effort` and `1.0`.
+
+In the example of the mobile manipulator, a possible configuration would look like the following:
+
+```
+Header header
+
+interface_name = ['wheel_left', 'wheel_right', 'joint_1', 'joint_2', …, 'joint_N', 'gripper_1']
+InterfaceValue = [wheel_left_iv, wheel_right_iv, joint_1_iv, joint_2_iv, …, joint_n_iv, gripper_1_iv]
+```
+where the `InterfaceValue` messages can be composed individually per `InterfaceName`.
+The name is similar to the already existing joint state message coming from the URDF.
+
+The `InterfaceValue` for the left wheel `wheel_left_iv` could be composed as such, reporting only its velocity:
+```
+value_identifier = ['velocity']
+value = [1.5]
+```
+
+The `InterfaceValue` for the first joint `joint_1_iv` reports a more complete interface:
+```
+value_identifier = ['position', 'velocity', 'effort']
+value = [1.1, 2.2, 3.3]
+```
+
+Additionally, the `InterfaceValue` for the gripper can be completely custom:
+```
+value_identifier = ['vaccuum_level', 'voltage']
+value = [4.4, 5.5]
+```
+
+That allows a relatively straight forward lookup of dynamically attached values per interface name.
+Each individual field is thus indexable by a tuple of `interface_name` and `value_identifier`.
+
+## Issues Considered
+
+### Name Lookup
+
+*Value Identifier Convention*
```

Comment:
We should enforce this through the URDF and transmission parsers. Design doc coming up. later this week.